src/stratis_cli/_actions/_top.py

```diff
@@ -204,20 +205,33 @@ def _add_update_key(proxy, key_desc, capture_key, *, keyfile_path):
     from ._data import Manager
 
     if capture_key:  # pragma: no cover
-        file_desc = sys.stdout.fileno()
-        fd_is_terminal = True
+        stdin_fd = sys.stdin.fileno()
+        (read, write) = os.pipe()
+        old_attrs = termios.tcgetattr(stdin_fd)
+        new_attrs = termios.tcgetattr(stdin_fd)
         print("Enter desired key data followed by the return key:")
+        new_attrs[3] = new_attrs[3] & ~termios.ECHO
+        termios.tcsetattr(stdin_fd, termios.TCSANOW, new_attrs)
+
+        # We can arbitrarily replace the newline with the empty string
+        # because readline() will only return a newline at the end of a
+        # buffered passphrase.
+        password = sys.stdin.readline().replace("\n", "")
```

Comment:
What about using `getpass`? https://docs.python.org/3.9/library/getpass.html

Reply:
Ok! Please give it a try: https://github.com/stratis-storage/stratis-cli/pull/674

Reply:
> Ok! Please give it a try: #674

The Cockpit PoC does not use this code path, it uses --keyfile-path. So I don't think there is much value in us testing this.